queries/yaml/highlights.scm

```diff
@@ -0,0 +1,29 @@
+(block_mapping_pair key: (flow_node) @field)
+(flow_pair key: (flow_node) @field) ; This one doesn't seem to work, no idea what's wrong with it.
```

Comment:
Did you tried looking in the playground to see if it captures what you want it to capture ?

Reply:
Yes, that's what I tried to do (that's how I came up with all of these!).

Reply:
specifically the playground tells me that I have something like this:
```
                      flow_mapping [34, 9] - [34, 22]
                        flow_pair [34, 11] - [34, 20]
                          key: flow_node [34, 11] - [34, 15]
                            plain_scalar [34, 11] - [34, 15]
                              string_scalar [34, 11] - [34, 15]
                          value: flow_node [34, 17] - [34, 20]
                            plain_scalar [34, 17] - [34, 20]
                              string_scalar [34, 17] - [34, 20]
```

Reply:
Could you test your query in the query tester of the playground too ?

Reply:
I'm not sure what this means - and I'm about to stop for the day.

If you can leave instructions I might get to it later in the week: or it should be super-easy to try yourself, just a yaml document containing something like `mapping: { foo: bar }` where we're trying to capture the foo.

Reply:
When using the playgroung, you can hit `o` to open a query debugger, that might be a good way to see why this query does not match anything.

Reply:
This could be a bug of the parser? It will never match that specific pattern and  when you use (_ key: flow_node) nvim freezes sometimes.

From the maker of tree-sitter-markdown. The complete parser is written in C.

Reply:
I don't know what the query debugger is supposed to do, it doesn't obviously do anything at all for me.

(Except, if I close the playground window containing the parse tree while the debugger is open, I get a segfault).

However, I found that I could get the highlighting to work by wildcarding:

```
(block_mapping_pair key: (_) @field)
(flow_mapping (_ key: (_) @field))
```

Here the `(_)` are just intended as an improvement / future-proofing on what I already had: let's treat any key as a `@field`.

But the first wilcard in the flow_mapping line is because I just don't know what to write there to make it work.  It seems to me that it should be `flow_pair`, but apparently it isn't.  Ho hum.

Reply:
I'm pretty sure that the parser causes the segfault or a incompatibility of the tree-sitter runtimes. When the maintainer creates his parser in C in can of course segfault.

Reply:
[video.webm.zip](https://github.com/nvim-treesitter/nvim-treesitter/files/5717304/video.webm.zip)

Should actually look like this? (I thought Github annouced video support? Could not upload the webm)